File: Baseball Game - Leetcode 682/Baseball Game - Leetcode 682.cpp
#include <vector>
#include <stack>
#include <string>
#include <sstream>

using namespace std;

class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> stk;

        for (const string& op : operations) {
            if (op == "+") {
                int top = stk.top(); stk.pop();
                int newTop = top + stk.top();
                stk.push(top);
                stk.push(newTop);
            } else if (op == "D") {
                stk.push(2 * stk.top());
            } else if (op == "C") {
                stk.pop();
            } else {
                stk.push(stoi(op));
            }
        }

        int sum = 0;
        while (!stk.empty()) {
            sum += stk.top(); stk.pop();
        }

        return sum;
    }
};

File: Baseball Game - Leetcode 682/Baseball Game - Leetcode 682.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    const stack = [];

    for (const op of operations) {
        if (op === "+") {
            const top = stack.pop();
            const newTop = top + stack[stack.length - 1];
            stack.push(top);
            stack.push(newTop);
        } else if (op === "D") {
            stack.push(stack[stack.length - 1] * 2);
        } else if (op === "C") {
            stack.pop();
        } else {
            stack.push(parseInt(op));
        }
    }

    return stack.reduce((acc, val) => acc + val, 0);
};
